### DISCUSSION TITLE
Mindstorms EV3: Advanced color sensoring functions detectable_colors() and hsv() not working

### DISCUSSION THREAD

**Discussion by rahulgitsit** [pybricks/support#952] [Q&A] [answered]
I am not able to use the ColorSensor methods such as detectable_colors() and hsv() for Mindstorms EV3. I am getting the following error while trying to use it. 
`AttributeError: 'ColorSensor' object has no attribute 'hsv'.`

I am able to use the functions like color() and reflection(). 

Sample code that I tried:

```
#!/usr/bin/env pybricks-micropython
from pybricks.hubs import EV3Brick
from pybricks.ev3devices import (
    Motor,
    TouchSensor,
    ColorSensor,
    InfraredSensor,
    UltrasonicSensor,
    GyroSensor,
)
from pybricks.parameters import Port, Stop, Direction, Button, Color
from pybricks.tools import wait, StopWatch, DataLog
from pybricks.robotics import DriveBase
from pybricks.media.ev3dev import SoundFile, ImageFile

# Create your objects here.
ev3 = EV3Brick()


# Write your program here.
# ev3.speaker.beep()

color_sensor = ColorSensor(Port.S3)
ultra = UltrasonicSensor(Port.S4)

my_colors = (Color.WHITE, Color.BLACK, Color.GREEN)
color_sensor.detectable_colors(my_colors)


while True:
    print("Distance: ", ultra.distance())
    print("Colour: ", color_sensor.color())
    print("HSV: ", color_sensor.hsv())
```

**Comment by laurensvalk** [comment, marked as answer]
Unfortunately, those methods are not available for the EV3 Color Sensor.

See [this page](https://pybricks.com/ev3-micropython/) for the EV3 documentation.

**Reply by rahulgitsit** [reply]
Oh! I am sorry,  I was in the wrong docs. Thank you!

**Reply by laurensvalk** [reply]
No worries. It's about time we start thinking about upgrading EV3 to the latest version of Pybricks to make things easier for everyone.

Even then, though, calibration may not be possible in the same form due to limitations of the EV3 Color sensor.
